### PR TITLE
Implement ticket activity page

### PR DESCRIPTION
**Pull request by song50119**
PR implement for ticket page activity
Screen Shot 2021-05-27 at 01 00 04

### CODE REVIEW

File: ui/tickets_activity_page.go [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+			},
+			body: func(gtx C) D {
+				walletID := c.info.Wallets[pg.walletDropDown.SelectedIndex()].ID
+				tickets := (*pg.tickets).Confirmed[walletID]
+				return layout.Stack{Alignment: layout.N}.Layout(gtx,
+					layout.Expanded(func(gtx C) D {
+						return layout.Inset{Top: values.MarginPadding60}.Layout(gtx, func(gtx C) D {
+							return c.theme.Card().Layout(gtx, func(gtx C) D {
+								gtx.Constraints.Min = gtx.Constraints.Max
+								if pg.ticketTypeDropDown.SelectedIndex()-1 != -1 {
```
Comment (oshorefueled) [inline ui/tickets_activity_page.go:68]: -1 in the if check is a magic number. Why is pg.ticketTypeDropDown being checked with "-1"

File: ui/tickets_activity_page.go [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+	}
+	return t
+}
+
+func (pg *ticketsActivityPage) handle() {
+	c := pg.common
+	initWalletDropDown(c, &pg.walletDropDown)
+
+	sortSelection := pg.orderDropDown.SelectedIndex()
+	if pg.filterSorter != sortSelection {
```
Comment (oshorefueled) [inline ui/tickets_activity_page.go:144]: Looks like this filterSorter check is for the purpose of only executing the ticket filter code when the dropdown is clicked. A better implementation is to have a Clicked() event emitted by the dropdown widget when the dropdown widget is clicked. You can look at how the Clicked() method is implemented in the clickable widget.

File: ui/tickets_activity_page.go [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+		}
+		return c.SubPageLayout(gtx, page)
+	}
+
+	return c.Layout(gtx, func(gtx C) D {
+		return c.UniformPadding(gtx, body)
+	})
+}
+
+func (pg *ticketsActivityPage) dropDowns(gtx layout.Context) layout.Dimensions {
```
Comment (oshorefueled) [inline ui/tickets_activity_page.go:102]: The wallet dropdown, sort dropdown and a page specific dropdown seems to be used in several pages. They could be implemented as "components" that can be called from each of those pages. This eliminates repetition.

File: ui/compontents.go [hunk 4]
```diff
@@ -960,6 +960,92 @@ func ticketCard(gtx layout.Context, c pageCommon, t *wallet.Ticket) layout.Dimen
 	})
 }
 
+// ticketRowActivity layouts out ticket info, display ticket activities on the tickets_page and tickets_activity_page
+func ticketRowActivity(gtx layout.Context, c *pageCommon, t wallet.Ticket, index int) layout.Dimensions {
```
Comment (oshorefueled) [inline ui/compontents.go]: This looks like it should be named `ticketActivityRow`

**Review by oshorefueled — CHANGES_REQUESTED** [review]

File: ui/tickets_activity_page.go (hunk 1 above)
Comment (song50119) [reply ui/tickets_activity_page.go:68]: Dropdowns is init with item + 1
`c := &DropDown{
		isOpen:         false,
		items:          make([]DropDownItem, len(items)+1), 
...`

**Review by oshorefueled — APPROVED** [review]

**Review by oshorefueled — APPROVED** [review]

**Pull request merged**